### PR TITLE
Fix interacting with selection on subplots with overlaid axes

### LINKED ISSUE #6530: Box Selection behaves weirdly with overlay = "y".
I have a scatter plot graph with two series, two y axis and an overlay = 'y'. 

The box selections seems not to behave correctly, when I do a first selection it selects correctly the points but if I extend the selection, the points are not  selected anymore. (see gif attachedl .

**You can find a reproducing example here : https://stackblitz.com/edit/web-platform-7oxvtx?file=index.html** 

plotly

### PR DESCRIPTION
**Pull request by Farkites**
Fixes #6530 

### Description 
Extending an existing box/lasso selection on traces that overlay axes don't update selection. Only the first trace (which doesn't have `overlay` is updated). See video below:
https://github.com/plotly/plotly.js/assets/35932204/91f75734-f47d-47b6-81b0-806413b2cf45

This was happening because the `yAxisIds` is equal to `'y'` and `trace.yaxis` to y + the index of the trace (e.g `y4`), so  `yAxisIds.indexOf(trace.yaxis)` is `-1` for all the overlaying traces.

### CODE REVIEW

File: src/components/selections/select.js [hunk 1]
```diff
@@ -809,8 +809,8 @@ function determineSearchTraces(gd, xAxes, yAxes, subplot) {
             var sankeyInfo = createSearchInfo(trace._module, cd, xAxes[0], yAxes[0]);
             searchTraces.push(sankeyInfo);
         } else {
-            if(xAxisIds.indexOf(trace.xaxis) === -1) continue;
-            if(yAxisIds.indexOf(trace.yaxis) === -1) continue;
+            if(xAxisIds.indexOf(trace.xaxis) === -1 && (!trace._xA || !trace._xA.overlaying)) continue;
+            if(yAxisIds.indexOf(trace.yaxis) === -1 && (!trace._yA || !trace._yA.overlaying)) continue;
```
Comment (archmoj) [inline src/components/selections/select.js:813]: Very interesting proposed fix.
It looks like we also select traces in other subplots that are not overlay with these axes.
If that's the case we shouldn't include them.

Comment (Farkites) [reply src/components/selections/select.js:813]: I have checked all the other test cases with subplots and the selection seems to be working fine. 

The only cases I've found where there are some crossover with selection/deselection are with:
- `plot_types`
- `plot_types_grid_dash`
- `box_plot_jitter_edge_cases`
- `mapbox_choropleth-multiple`

And for these the issues happen with and without my changes.

Comment (archmoj) [reply src/components/selections/select.js:813]: @Farkites Thanks very much for your tests.
Here we need to know if we have two subplots i.e. (xy + xy2 ) and (x2y3 + x2,y4) and the user reselects over the first one; then are the traces in the second subplot (x2y3) are also re-selected before and after this change?

Comment (Farkites) [reply src/components/selections/select.js:813]: @archmoj Yes, in that scenario the traces in the second subplot (x2y3) are re-selected before and after the changes. You can check this here:
- Before changes: https://codepen.io/Mario-Rodr-guez-Ib-ez/pen/ExMEbdK
- After changes: https://codepen.io/Mario-Rodr-guez-Ib-ez/pen/QWoQoao

**Comment by archmoj** [comment]
Thanks very much for all these manual tests and the brilliant fix. 
I am tempted to include this in the next plotly.js minor which is coming out today.
So I'll approve it now.

:dancer: :dancer:

Please open a follow up PR to include jasmine tests to lock down the bug.
You may place new tests in `test/jasmine/tests/select_test.js`
Thank you!

**Comment by archmoj** [comment]
@Farkites One more request before merging this.
Please create a draft log file and thanks the sponsors.

**Pull request merged**